ocaml/dune

```diff
@@ -60,13 +72,19 @@
    ; remove -w -67 by adding the camlinternalMenhirLib hack like the Makefile
  ))
  (ocamlopt_flags (:include %{project_root}/ocamlopt_flags.sexp))
+ (preprocessor_deps parser_types.ml parser_types.mli)
  (preprocess
-   (per_module
-     ((action
-        (progn
-          (echo "module MenhirLib = CamlinternalMenhirLib")
-          (run cat %{input-file})))
-       parser)))
+  (per_module
+   ((action
+     (progn
+      (system
+      ; Copy over the interface/implementation of parser_types to prevent the preprocessed
+      ; autogenerated Menhir files from having unbound types/modules exposed from the
+      ; Menhir inspection interface.
+       "[ $(basename %{input-file} | sed 's/.[^.]*$//').mli = $(basename %{input-file}) ] && cat ocaml/parser_types.mli || cat ocaml/parser_types.ml")
+      (system "echo 'module MenhirLib = CamlinternalMenhirLib'")
+      (run cat %{input-file})))
```

Comment:
It's not obvious why you need to do any copying. Why doesn't the parser just refer to `Parser_types` as an ordinary module? The copied text includes references to other such modules e.g. `Asttypes`.

Reply:
I just tried this out, and actually Menhir does properly generate the .mli file without this line altogether, so I can actually remove it altogether. (Moving the types to parser_types.ml caused the inspection API interface to refer to the previously-problematic types as Parser_types.t)